### PR TITLE
  EOS-14447 scripts/m0provision: update motr config post rpm installation

### PR DESCRIPTION
**Pull request by madhavemuri**
Problem:
    Currently provisioner doesn't provide the interface to update config
    post the rpm updates.

    Solution:
    Use m0provision to update the motr.conf specific to hardware once rpm is
    installed.

### CODE REVIEW

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98/>Motr#98</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>32m0t1fs-rconfc-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/47motr-tests-user-kernel target=_blank>47motr-tests-user-kernel</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/31motr-sys-kvs-kernel-test target=_blank>31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/07multi-client-tests target=_blank>07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/39mount-fail target=_blank>39mount-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/12sns-cc-io-tests target=_blank>12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/37protocol target=_blank>37protocol</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>56</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/40motr-dgmode target=_blank>40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/51motr-rmw target=_blank>51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/50motr-rm-lock-cc-io target=_blank>50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/20pool-version-test target=_blank>20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/46m0crate target=_blank>46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/54sns-repair-motr-1f target=_blank>54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/25spiel-sns-repair-test target=_blank>25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/21spiel-test target=_blank>21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/59sns-repair-motr-abort target=_blank>59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/01kernel-tests target=_blank>01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/33motr-st target=_blank>33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/42motr-utils target=_blank>42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/19sns-abort-test target=_blank>19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/09poolmach-tests target=_blank>09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/13sns-repair-quiesce target=_blank>13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/41motr-conf-update target=_blank>41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/17sss-st-test target=_blank>17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/00userspace-tests target=_blank>00userspace-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/03initscript-tests target=_blank>03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/45motr-sns-repair-N-1 target=_blank>45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/30hagen-test target=_blank>30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/11sns-multi-tests target=_blank>11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/58sns-repair-motr-ios-fail target=_blank>58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/29confgen-test target=_blank>29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/34sns-repair-1n-1f target=_blank>34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/44motr-sns-repair target=_blank>44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/08degraded-mode-tests target=_blank>08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/35m0singlenode target=_blank>35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/14m0t1fs-fsync-test target=_blank>14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/43motr-sync-replication target=_blank>43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/04net-tests target=_blank>04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/55sns-repair-motr-1n-1f target=_blank>55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/24m0d-fsync-test target=_blank>24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/49motr-rpc-cancel target=_blank>49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/56sns-repair-motr-mf target=_blank>56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/38sns-abort-quiesce-test target=_blank>38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/15m0t1fs-fwait-test target=_blank>15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/02system-tests target=_blank>02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/22rpc-cancel-test target=_blank>22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/53clusterusage-alert target=_blank>53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/05rpcping-tests target=_blank>05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/23m0d-signal-test target=_blank>23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/28motr-sys-kvs-test target=_blank>28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/06console-tests target=_blank>06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/61spiel-multi-confd target=_blank>61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/18m0mount-test target=_blank>18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/27sns-repair-io-fail-test target=_blank>27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/16conf-st-test target=_blank>16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/52singlenode-sanity target=_blank>52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/57sns-repair-motr-1k-1f target=_blank>57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/48motr-raid0-io target=_blank>48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/10sns-single-tests target=_blank>10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/36spare-reservation target=_blank>36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node/51kem target=_blank>51kem</a></p></details></td></tr><tr><td>Total</td><td>63</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/98//testReport/junit/motr-single-node>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

File: scripts/install/opt/seagate/cortx/motr/conf/setup.yaml [hunk 1]
```diff
@@ -29,7 +29,6 @@ motr:
     config:
         script: /usr/sbin/m0provision
         args:
-            - config
```
Comment (yatin-mahajan) [inline scripts/install/opt/seagate/cortx/motr/conf/setup.yaml:32]: [question]
If we have removed args : - config then does it make sense keeping m0provision with empty arguments ? can we remove this whole block ?

File: scripts/install/opt/seagate/cortx/motr/conf/setup.yaml [hunk 2]
```diff
@@ -45,7 +44,6 @@ motr:
     post_update:
         script: /usr/sbin/m0provision
         args:
-            - config
```
Comment (yatin-mahajan) [inline scripts/install/opt/seagate/cortx/motr/conf/setup.yaml:48]: Same here, do we need this block as ha is not going to call post_update and we are handling it in rpm update %postinstall tag ?

File: scripts/install/opt/seagate/cortx/motr/libexec/motr_cfg.sh [hunk 3]
```diff
@@ -165,7 +165,7 @@ do_m0provision_action()
     msg "Configuring host [`hostname -f`]"
 
     MD_DEVICE=$(lvs -o lv_path | grep "lv_raw_metadata" | head -1)
-    if [[ $? -eq 0 ]];then
+    if [[ $? -eq 0 && $MD_DEVICE != "" ]];then
```
Comment (yatin-mahajan) [inline scripts/install/opt/seagate/cortx/motr/libexec/motr_cfg.sh:168]: [suggestion]
Can also be replaced by -n "$MD_DEVICE"

**Review by yatin-mahajan — COMMENTED** [review]
Few question.

File: scripts/install/opt/seagate/cortx/motr/conf/setup.yaml (hunk 1 above)
Comment (madhavemuri) [reply scripts/install/opt/seagate/cortx/motr/conf/setup.yaml:32]: This is kept for future, i.e after GA or for LDR R2

Comment (madhavemuri) [reply scripts/install/opt/seagate/cortx/motr/conf/setup.yaml:32]: Provisioner calls config section after deployment, but not after updates, thus i am making it a no-op here.

**Review by yatin-mahajan — APPROVED** [review]

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102/>Motr#102</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>3</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/32m0t1fs-rconfc-fail-test target=_blank>32m0t1fs-rconfc-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/23m0d-signal-test target=_blank>23m0d-signal-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/00userspace-tests target=_blank>00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>5</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/31motr-sys-kvs-kernel-test target=_blank>31motr-sys-kvs-kernel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/07multi-client-tests target=_blank>07multi-client-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/39mount-fail target=_blank>39mount-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/12sns-cc-io-tests target=_blank>12sns-cc-io-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/37protocol target=_blank>37protocol</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>55</td><td><details><summary>:file_folder:</summary><p><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/40motr-dgmode target=_blank>40motr-dgmode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/51motr-rmw target=_blank>51motr-rmw</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/50motr-rm-lock-cc-io target=_blank>50motr-rm-lock-cc-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/20pool-version-test target=_blank>20pool-version-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/46m0crate target=_blank>46m0crate</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/54sns-repair-motr-1f target=_blank>54sns-repair-motr-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/25spiel-sns-repair-test target=_blank>25spiel-sns-repair-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/21spiel-test target=_blank>21spiel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/59sns-repair-motr-abort target=_blank>59sns-repair-motr-abort</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/01kernel-tests target=_blank>01kernel-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/33motr-st target=_blank>33motr-st</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/42motr-utils target=_blank>42motr-utils</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/47motr-tests-user-kernel target=_blank>47motr-tests-user-kernel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/19sns-abort-test target=_blank>19sns-abort-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/09poolmach-tests target=_blank>09poolmach-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/13sns-repair-quiesce target=_blank>13sns-repair-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/41motr-conf-update target=_blank>41motr-conf-update</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/17sss-st-test target=_blank>17sss-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/03initscript-tests target=_blank>03initscript-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/45motr-sns-repair-N-1 target=_blank>45motr-sns-repair-N-1</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/30hagen-test target=_blank>30hagen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/11sns-multi-tests target=_blank>11sns-multi-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/58sns-repair-motr-ios-fail target=_blank>58sns-repair-motr-ios-fail</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/29confgen-test target=_blank>29confgen-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/34sns-repair-1n-1f target=_blank>34sns-repair-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/44motr-sns-repair target=_blank>44motr-sns-repair</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/08degraded-mode-tests target=_blank>08degraded-mode-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/26spiel-sns-repair-quiesce-test target=_blank>26spiel-sns-repair-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/60sns-repair-motr-abort-quiesce target=_blank>60sns-repair-motr-abort-quiesce</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/35m0singlenode target=_blank>35m0singlenode</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/14m0t1fs-fsync-test target=_blank>14m0t1fs-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/43motr-sync-replication target=_blank>43motr-sync-replication</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/04net-tests target=_blank>04net-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/55sns-repair-motr-1n-1f target=_blank>55sns-repair-motr-1n-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/24m0d-fsync-test target=_blank>24m0d-fsync-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/49motr-rpc-cancel target=_blank>49motr-rpc-cancel</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/56sns-repair-motr-mf target=_blank>56sns-repair-motr-mf</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/38sns-abort-quiesce-test target=_blank>38sns-abort-quiesce-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/15m0t1fs-fwait-test target=_blank>15m0t1fs-fwait-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/02system-tests target=_blank>02system-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/22rpc-cancel-test target=_blank>22rpc-cancel-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/53clusterusage-alert target=_blank>53clusterusage-alert</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/05rpcping-tests target=_blank>05rpcping-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/28motr-sys-kvs-test target=_blank>28motr-sys-kvs-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/06console-tests target=_blank>06console-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/61spiel-multi-confd target=_blank>61spiel-multi-confd</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/18m0mount-test target=_blank>18m0mount-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/27sns-repair-io-fail-test target=_blank>27sns-repair-io-fail-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/16conf-st-test target=_blank>16conf-st-test</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/52singlenode-sanity target=_blank>52singlenode-sanity</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/57sns-repair-motr-1k-1f target=_blank>57sns-repair-motr-1k-1f</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/48motr-raid0-io target=_blank>48motr-raid0-io</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/10sns-single-tests target=_blank>10sns-single-tests</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/36spare-reservation target=_blank>36spare-reservation</a></br><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node/51kem target=_blank>51kem</a></p></details></td></tr><tr><td>Total</td><td>63</td><td><a href=http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/102//testReport/junit/motr-single-node>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

File: scripts/install/opt/seagate/cortx/motr/libexec/motr_cfg.sh [hunk 4]
```diff
@@ -179,16 +179,19 @@ do_m0provision_action()
...
+    platform=$(get_platform)
+    echo "Server type is $platform"
 
     ANY_ERR=$(echo $SALT_OPT | grep -i ERROR | wc -l)
     if [ "$ANY_ERR" != "0" ]; then
-        err "Salt command failed."
+        msg "Salt command failed or not available."
         msg "[$SALT_OPT]"
-        die $ERR_CFG_SALT_FAILED
+        #die $ERR_CFG_SALT_FAILED
```
Comment (huanghua78) [inline scripts/install/opt/seagate/cortx/motr/libexec/motr_cfg.sh:189]: Why is this commented out?
Why don't we exit with error?

Comment (madhavemuri) [reply scripts/install/opt/seagate/cortx/motr/libexec/motr_cfg.sh:189]: If salt is not available and motr rpm is installed on any physical machine, these changes are applied, see line number 194.

**Review by huanghua78 — APPROVED** [review]

**Pull request merged**